### ISSUE TITLE
Support for DXVK on windows/loading custom d3d9.dll

### ISSUE THREAD

**Issue by Aib0t** [tobii-dev/blur-hooks-rs#4]
```
Hi, update 2.5 causes a problem for me:
The AMD GPU driver for 7000 series has a bug that causes multi-second freezes when you drive through shunt distortion trail.
Until 2.5 I could work around the bug by using DXVK translation DLL for D3D9. Since AMAX 2.5 also replaces D3D9.dll this no longer works.
Is there a way to make AMAX's D3D9.dll load another custom D3D9.dll like DXVK?
```

**Comment by tobii-dev** [comment]
Move the DXVK d3d9.dll to amax/dlls/d3d9.dll
This works for ReShade, and might also work for DXVK. Otherwise, we will need to find a way to load DXVK before any d3d9 stuff loads.

**Comment by tobii-dev** [comment]
It will need some additional work, but I think it's doable:
https://github.com/doitsujin/dxvk/issues/2772

**Comment by tobii-dev** [comment]
After some testing:

- Loading it the same way as ReShade doesn't work, unfortunately. The reason is that ReShade just forwards de calls to the real system `d3d9.dll`.

- Replacing `C:\Windows\SysWOW64\d3d9.dll` with the DXVK `d3d9.dll` works, as long as egui-d3d9 doesn't draw anything. However this is a very ugly solution.

- Forwarding calls from Blur `d3d9.dll` to the DXVK `d3d9.dll` works, but egui-d3d9 crashes. I can live with that, the egui window is disabled for now anyways.

To add support for DXVK, what I'll do is the following:

1. Try to load `amax/dxvk.dll` on init.
2. If it loaded, forward any D3D9 calls to their DXVK implementation. If it didn't load, forward to system `d3d9.dll` like we have been doing until now.

It could also be done with feature flags but I don't wanna create multiple releases. The only real performance loss is on init, and its minimal, so it should be fine.

**Comment by tobii-dev** [comment]
Fixed in 753362071dd6df9f663c4f675ae99f4ccfd57ff0
plz test

**Comment by mdro** [comment]
OP here, could you upload a build? I don't have Rust set up rn.
Also noticed something strange while testing with amax/dlls: If I add DXVK's `d3d10core.dll` (and optionally dxgi and d3d11) to the folder I get a crash on startup: It's looking for the entry point `D3D11CoreCreateDevice@24`. 1. What wants to use D3D10/11 here? 2. looking for D3D11 in D3D10 DLL looks like it would only happen to work due to implementation details.

**Comment by tobii-dev** [comment]
> upload a build?

https://github.com/tobii-dev/blur-hooks-rs/suites/16593808120/artifacts/948015

[blur-hook-rs_release.zip](https://github.com/tobii-dev/blur-hooks-rs/files/12732252/blur-hook-rs_release.zip)

> with amax/dlls: If I add DXVK's `d3d10core.dll` (and optionally dxgi and d3d11) to the folder I get a crash on startup: It's looking for the entry point `D3D11CoreCreateDevice@24`.

We try to load ANY `.asi` or `.dll` files in `amax/dlls/`

> 1. What wants to use D3D10/11 here?

AFAIK, Blur only tries to work with DirectX9. Maybe some funky windows stuff trying make use of more modern DirectX versions supported by the GPU?

Otherwise, and what I think happens: DXVK itself tries to load it: https://github.com/doitsujin/dxvk/blob/494f7fd38df9b0ec6b1d3137a6528d2f31d55452/src/d3d10/d3d10_core.cpp#L32C5-L32C5
I don't know why though.

> 2. looking for D3D11 in D3D10 DLL looks like it would only happen to work due to implementation details.

Then it is probably DXVK implementation details, but I don't know why exactly they use D3D11 from D3d10.
Here is an old commit on DXVK D3D10: https://github.com/doitsujin/dxvk/blob/494f7fd38df9b0ec6b1d3137a6528d2f31d55452/src/d3d10/d3d10_core.cpp#L32C5-L32C5


It should work by just having `amax/dxvk.dll`.
Please let me know!

**Comment by mdro** [comment]
It works! Thanks a lot!
Since it seems to be DXVK internal that D3D11 call shouldn't be a problem.

**Comment by tobii-dev** [comment]
Fixed in 0.2.1 - https://github.com/tobii-dev/blur-hooks-rs/commit/753362071dd6df9f663c4f675ae99f4ccfd57ff0
